config/src/config.rs

```diff
@@ -68,12 +68,16 @@ impl ConfigPrefix {
 }
 
 impl Config {
-    pub fn parse(prefix: ConfigPrefix, filename: &PathBuf) -> Result<Self, String> {
+    pub fn parse(prefix: ConfigPrefix, filename: &Option<PathBuf>) -> Result<Self, String> {
```

Comment:
Almost always it's not a good idea to pass a reference to an Option. In case you want a reference, you want to pass an Option to a reference.

`&Option<T>` -> `Option<&T>`.